packages/dialog/src/lit/renderer-directives.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+/* eslint-disable max-classes-per-file */
+import { directive } from 'lit/directive.js';
+import { microTask } from '@vaadin/component-base/src/async.js';
+import { Debouncer } from '@vaadin/component-base/src/debounce.js';
+import { LitRendererDirective } from '@vaadin/lit-renderer';
+
+const CONTENT_UPDATE_DEBOUNCER = Symbol('contentUpdateDebouncer');
+
+class AbstractDialogRendererDirective extends LitRendererDirective {
+  /**
+   * A property to that the renderer callback will be assigned.
+   *
+   * @abstract
+   */
+  rendererProperty;
+
+  /**
+   * Adds the renderer callback to the dialog.
+   */
+  addRenderer() {
+    this.element[this.rendererProperty] = (root, dialog) => {
+      this.renderRenderer(root, dialog);
+    };
+  }
+
+  /**
+   * Runs the renderer callback on the dialog.
+   */
+  runRenderer() {
+    this.element[CONTENT_UPDATE_DEBOUNCER] = Debouncer.debounce(
+      this.element[CONTENT_UPDATE_DEBOUNCER],
+      microTask,
+      () => {
+        this.element.requestContentUpdate();
+      },
+    );
+  }
+
+  /**
+   * Removes the renderer callback from the dialog.
+   */
+  removeRenderer() {
+    this.element[this.rendererProperty] = null;
```

Comment:
Nit: Could also remove `this.element[CONTENT_UPDATE_DEBOUNCER]`

Reply:
Done.